ev3/legoev3_motor.c

```diff
@@ -819,6 +830,9 @@ static int legoev3_motor_get_state(void *context)
 			state |= BIT(TM_STATE_OVERLOADED);
 	}
 
+	if (state != original)
+		tacho_motor_notify_state_change(&ev3_tm->tm);
```

Comment:
State does not change here. `legoev3_motor_get_state()` is translating `STATE_*` to `TM_STATE_*` flags. These are two different enums and one is used as bit flags, so comparing `state` to `ev3_tm->state` (via `original`) is not valid anyway.